Feature/compiler

virtustron opened the pull request.

virtustron committed:
Add package information to Compiler/Cargo.toml

AndriiSerbeniuk reviewed (dismissed):
Looks great

virtustron committed:
Add basic files to compiler crate

virtustron committed:
Trying to compile cpp

virtustron committed:
Add g++ compilation of cpp-file

virtustron committed:
Add todo

virtustron committed:
Make general structure of all modules

virtustron committed:
Add trait compiler

virtustron committed:
Fix Compiler trait implementation and usage

virtustron committed:
Simplify in and out data structures, fix errors and warnings

virtustron committed:
Change OutputData structure

virtustron committed:
Implement new out data structures in compile fun

virtustron committed:
Add handler::run_compilation usage to main fun

virtustron committed:
Add temp folder for source and compiled files

virtustron committed:
Implement compilation specific source file

virtustron committed:
Add filename generation by DateTime value

virtustron committed:
Write TODO list

virtustron committed:
Add compiler_options to input_data and its usage

virtustron committed:
Moving to Path

virtustron committed:
Use PathBuf type instead of Path

virtustron committed:
Add path to src and bin files to in and out data

virtustron committed:
Delete unnecessary bin file

virtustron committed:
Add constructor to struct InputData

virtustron committed:
Add doc comment to run_compilation

virtustron committed:
Change stdout and stderr type to String

OleksandrStrashnenko commented on compiler/src/compilers/cpp_compiler.rs:
```diff
@@ -0,0 +1,47 @@
+        full_bin_filename.push(&bin_file_name);
+
+        let compiler_output = Command::new("g++")
+                .args(&[&input_data.compiler_options, 
```
There is a problem with compiler_options containing "".  
g++ takes "" as a file name.

virtustron replied:
Thanks a lot! Fix this problem - add if statement

virtustron committed:
Fix bug with epty compiler_options in InputData

virtustron committed:
Add doc comments

virtustron committed:
Add TODO records

virtustron committed:
Changed "compile" fn return type to Result<> (#15)

OleksandrStrashnenko committed:
Implement Compiler trait for RustCompiler.

Use /dev/null as binary file name, this is temporary decision, while binary is unused.
Implement new method for OutputData, that return new empty OutputData instance.

virtustron committed:
Merge pull request #16 from ITA-Dnipro/feature/compiler-rust

Implement Compiler trait for RustCompiler.

virtustron committed:
Add "compilation code" string to stdout

valentynfaychuk commented on compiler/src/compilers/compiler.rs:
```diff
@@ -0,0 +1,7 @@
+
+
+pub trait Compiler {
+    fn compile(&self, input_data: &InputData) -> Result<OutputData, &'static str>;
```
Seems like calls to this function can block for a considerable amount of time, can this function be `async`?

valentynfaychuk commented on compiler/src/storage/name_generator.rs:
```diff
@@ -0,0 +1,26 @@
+    let now = chrono::Utc::now();
+    let filename = String::from(now.format("file-%Y-%m-%d-%H-%M-%S-%f").to_string());
+    filename
+}
```
Can we have "newlines" at the end of the files?
